src/lib/services/moduleService.ts

```diff
@@ -98,3 +107,47 @@ export const useVerifyModule = ({
       keepPreviousData: true,
     }
   );
+
+export interface DecodeModuleQueryResponse {
+  abi: ResponseABI;
+  modulePath: string;
+  validPublisher: boolean;
+  currentPolicy: Option<UpgradePolicy>;
+}
+
+export const useDecodeModule = ({
+  moduleEncode,
```

Comment:
I suggest renaming it to base64EncodeFile. This name is more descriptive and meaningful.